src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -166,44 +168,59 @@ inline TShape ReduceAxesShapeImpl(const TShape& ishape, const TShape& axis,
     }
   }
 
+  if (axis[0] < 0) {
```

Comment:
axis is not sorted now. These two checks should be done later.
Also remove the sort in attr_parser